keyboards/divergetm2/keymaps/xtonhasvim/keymap.c

```diff
@@ -0,0 +1,195 @@
+ * |--------+------+------+------+------+-------------+------+------+------+------+------|
+ * |BL Lower|      |      |      |      |      |      |      |      |      |      |      |
+ * |--------+------+------+------+------+------|------+------+------+------+------+------|
+ * |BL STEP |      |      |      |      |      |      | Next | Vol- | Vol+ | Play |      |
+ * |--------+------+------+------+------+------+------+------+------+------+------+------|
+ * |Backlite| Bail |      |      |             |             |      |      | Bail |      |
+ * `-------------------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_ortho_4x12_2x2u( \
+  BL_INC, X_____X,   X_____X,   X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, \
+  BL_DEC, X_____X, X_____X,  X_____X,   X_____X,  X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, \
+  BL_STEP, X_____X,  X_____X,  X_____X,  X_____X,  X_____X, X_____X,  KC_MNXT, KC_VOLD, KC_VOLU, KC_MPLY, X_____X, \
+  BL_TOGG, TO(_MOUSE), _______, _______, _______,                 _______, _______, _______, TO(_QWERTY), X_____X \
+),
+
+
+/* movement layer (hold semicolon) */
+[_MOVE] = LAYOUT_ortho_4x12_2x2u( \
+  TO(_QWERTY), X_____X, X_____X, X_____X, X_____X, X_____X, KC_HOME, KC_PGDN, KC_PGUP, KC_END, X_____X, X_____X, \
+  _______,     X_____X, LGUI(KC_LBRC), LGUI(LSFT(KC_LBRC)), LGUI(LSFT(KC_RBRC)), LGUI(KC_RBRC), KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, _______, X_____X, \
+  _______,     X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, _______, \
+  _______,     _______, _______, _______, _______,                   _______, _______, _______, _______, X_____X \
+),
+
+/* mouse layer
+ */
+[_MOUSE] = LAYOUT_ortho_4x12_2x2u( \
+  TO(_QWERTY), X_____X, X_____X, KC_MS_UP, X_____X, X_____X, KC_MS_WH_LEFT, KC_MS_WH_DOWN, KC_MS_WH_UP, KC_MS_WH_RIGHT, X_____X, X_____X  , \
+  _______,     X_____X, KC_MS_LEFT, KC_MS_DOWN, KC_MS_RIGHT, X_____X, X_____X, KC_MS_BTN1, KC_MS_BTN2, KC_MS_BTN3, X_____X, X_____X, \
+  _______,     X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, X_____X, _______, \
+  _______,     TO(_QWERTY), _______, _______, _______,                    _______, _______, _______, TO(_QWERTY), X_____X \
+),
+
+/* vim command layer.
+ */
+[_CMD] = LAYOUT_ortho_4x12_2x2u( \
+  X_____X, X_____X, VIM_W, VIM_E, X_____X, X_____X, VIM_Y, VIM_U, VIM_I, VIM_O, VIM_P, X_____X, \
+  VIM_ESC,    VIM_A, VIM_S, VIM_D, X_____X, VIM_G, VIM_H, VIM_J, VIM_K, VIM_L, X_____X, X_____X, \
+  VIM_SHIFT,     X_____X, VIM_X, VIM_C, VIM_V, VIM_B, X_____X, X_____X, VIM_COMMA, VIM_PERIOD, X_____X, VIM_SHIFT, \
+  _______,     TO(_QWERTY), _______, _______, X_____X,                 X_____X, _______, _______, TO(_QWERTY), X_____X \
+)
+
+};
+
+
+#define SPACE_WAIT 100
+uint16_t rl_start_time = 0;
+
+
+bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
```

Comment:
You could replace all of this with just: 

```c
#define RAISE LT(_RAISE, KC_SPC
#define LOWER LT(_LOWER, KC_SPC

uint32_t layer_state_set_user(uint32_t state) {
  return update_tri_layer_state(state, _RAISE, _LOWER, _ADJUST);
}
```

And it would work better, I think.